internal/cmd/certgen.go

```diff
@@ -35,7 +35,7 @@ func getCertGenCommand() *cobra.Command {
 
 // certGen generates control plane certificates.
 func certGen() error {
-	cfg, err := getConfig()
+	cfg, err := getConfig(cfgPath)
```

Comment:
looks like cfgPath is a global var in the `cmd` package, dont understand the issue here

Reply:
In getConfig test, I need to create a temp file and use the file path to get the config. I add a argument to `getConfig` so that I can pass the temp file path to `getConfig`. In this way, the global `cfgPath` won't be polluted by the test.

Reply:
Or maybe I can define a new function like `getConfigByPath()` and call it in `getConfig` with the global var. Then I can test `getConfigByPath` in test and the cmd package can still use `getConfig`.

Reply:
@arkodg Any idea?

Reply:
can you help understand what this PR is `fixing`, or is it just increasing test coverage for the package ?

Reply:
`cfg.Validate()` is called just after `config.New()` before this fix and the user config won't be validated if `cfgPath` is provided via `--config-path`. After the fix, `getConfig` will return an error if user uses an invalid config.

Reply:
why not leave the current API as is, and just set the global var `cfgPath` in the Test ?

Reply:
> why not leave the current API as is, and just set the global var `cfgPath` in the Test ?

I just don't want to pollute the global var in the tests(maybe due to my unimportant habit in functional languages XD), and this may cause other tests to fail if `cfgPath` is also used in other tests in the future. Yeah, I can do this and recover it after the test, this is fine in theory.

Reply:
can you delete the global var `cfgPath`, make it local to the func and instead pass by value everywhere ?

Reply:
I think we can still keep it global now because it's used as a flag and it will be hard to use in non-test code. We can do this after some refactor like wrapping all flags in a struct then change functions like `server()` to its methods. For this PR, I think we can just do what I said before:

> Or maybe I can define a new function like getConfigByPath() and call it in getConfig with the global var. Then I can test getConfigByPath in test and the cmd package can still use getConfig.

btw, I found another small thing: `getConfig` is used in `certGen`, but `cfgPath` is not registered as a flag. I'm not sure if this is a problem.

Reply:
I updated the code as I said. Please let me know if there's anything you feel not good. Thanks!

Reply:
the changes are not ideal, but fine for this PR (would have ideally liked to get rid of the global var so it doesnt confuse the next dev)

Reply:
but looks like the way the code is today, with `certGen` calling `getConfig`, its probably easier to keep it as a global var.
thanks again for taking the time to explain the fix and addressing comments